src/Integration.UnitTests/NewConnectedMode/ConfigurationProviderTests.cs

```diff
@@ -141,37 +209,55 @@ public void WriteConfig_StandaloneConfig_Throws()
         }
 
         [TestMethod]
-        public void WriteConfig_LegacyConfig_CallsLegacySerializer()
+        public void WriteConfig_LegacyConfig_SavesLegacyConfig()
         {
             // Arrange
-            legacySerializer.WriteSolutionBindingAction = p => "filename.txt";
+            var config = BindingConfiguration.CreateBoundConfiguration(new BoundSonarQubeProject(), SonarLintMode.LegacyConnected);
 
-            var config = BindingConfiguration.CreateBoundConfiguration(new BoundSonarQubeProject(), isLegacy: true);
+            legacySerializer.Setup(x => x.Get()).Returns("old.txt");
+            solutionBindingFile.Setup(x => x.ReadSolutionBinding("old.txt")).Returns(new BoundSonarQubeProject());
+
+            solutionBindingFile
+                .Setup(x => x.WriteSolutionBinding("old.txt", config.Project, legacyPostSaveOperation.Object.OnSuccessfulSave))
+                .Returns(true);
 
             // Act
             var actual = testSubject.WriteConfiguration(config);
 
             // Assert
             actual.Should().BeTrue();
-            legacySerializer.WrittenFilesCount.Should().Be(1);
-            newSerializer.WrittenFilesCount.Should().Be(0);
+
+            solutionBindingFile.Verify(x =>
+                    x.WriteSolutionBinding("old.txt", config.Project, legacyPostSaveOperation.Object.OnSuccessfulSave),
+                Times.Once);
         }
 
         [TestMethod]
-        public void WriteConfig_NewConnectedModeConfig_CallsNewSerializer()
+        public void WriteConfig_NewConnectedModeConfig_SaveNewConfig()
         {
-            // Arrange
-            newSerializer.WriteSolutionBindingAction = p => "filename.txt";
+            var projectToWrite = new BoundSonarQubeProject();
+            var config = BindingConfiguration.CreateBoundConfiguration(projectToWrite, SonarLintMode.Connected);
 
-            var config = BindingConfiguration.CreateBoundConfiguration(new BoundSonarQubeProject(), isLegacy: false);
+            legacySerializer.Setup(x => x.Get()).Returns(null as string);
+            newSerializer.Setup(x => x.Get()).Returns("new.txt");
+
+            solutionBindingFile
```

Comment:
This doesn't feel right; the test shouldn't need to mock the existence of a file for the WriteConfig operation to work.

When a solution is bound for the first time, the config file won't exist, so the first call to WriteConfig should create it.

Reply:
Yep you're right